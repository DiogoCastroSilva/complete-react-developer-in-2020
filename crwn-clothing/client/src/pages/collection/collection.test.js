import React from 'react';
import { shallow } from 'enzyme';

import Collection from './collection.component';
import CollectionItemContainer from '../../containers/collection-item.container';


describe('testing Collection component', () => {
    describe('Collection without items', () => {
        let wrapper;

        beforeEach(() => {
            const mockProps = {
                collection: {
                    id: 1,
                    title: 'hats',
                    items: []
                }
            };
    
            wrapper = shallow(<Collection {...mockProps} />);
        });
    
        it('should render a Collection component', () => {
            expect(wrapper).toMatchSnapshot();
        });
    
        it('should render 0 items', () => {
            expect(wrapper.find(CollectionItemContainer).exists()).toBe(false);
        });
    });
    

    it('should render 2 item', () => {
        const mockProps = {
            collection: {
                title: 'hats',
                items: [
                    {
                        id: 1,
                        imageUrl: 'https://store.storeimages.cdn-apple.com/4668/as-images.apple.com/is/iphone-12-pro-family-hero?wid=940&amp;hei=1112&amp;fmt=jpeg&amp;qlt=80&amp;op_usm=0.5,0.5&amp;.v=1604021663000',
                        price: '1179€',
                        name: 'Iphone 12 Pro'    
                    },
                    {
                        id: 2,
                        imageUrl: 'https://store.storeimages.cdn-apple.com/4668/as-images.apple.com/is/iphone-12-pro-family-hero?wid=940&amp;hei=1112&amp;fmt=jpeg&amp;qlt=80&amp;op_usm=0.5,0.5&amp;.v=1604021663000',
                        price: '179€',
                        name: 'SSD driver'
                    }
                ]
            }
        };

        const wrapper = shallow(<Collection {...mockProps} />);
        expect(wrapper.find(CollectionItemContainer).exists()).toBe(true);
        expect(wrapper.find(CollectionItemContainer).length).toBe(mockProps.collection.items.length);
    });
});