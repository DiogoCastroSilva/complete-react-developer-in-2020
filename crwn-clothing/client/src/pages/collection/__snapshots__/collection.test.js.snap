// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Collection component Collection without items should render a Collection component 1`] = `
<styled.div>
  <styled.h2>
    hats
  </styled.h2>
  <styled.div />
</styled.div>
`;
